scripts/maintenance/generate_audit_summary.py

```
@@ -0,0 +1,118 @@
+def run_command(args):
+    """Executes a command securely using list-based arguments and shell=False."""
+    try:
+        result = subprocess.run(args, capture_output=True, text=True, check=True)
+        return result.stdout.strip()
+    except subprocess.CalledProcessError as e:
+        # Don't log errors for 'which' checks
+        if args[0] != 'which':
+            print(f"Error: {e.stderr}", file=sys.stderr)
+        return None
+
+def main():
```

Ruff (0.15.9)

[warning] Too many branches (21 > 12)

(PLR0912)